src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java

```diff
@@ -76,10 +78,29 @@ public Artefact createPublication(Artefact artefact, MultipartFile file) {
             file
         );
         artefact.setPayload(blobUrl);
-
         return artefactRepository.save(artefact);
     }
 
+    /**
+     * Checks if the artefact has a display from date of today or previous then triggers the sub fulfilment
+     * process on subscription-management if appropriate.
+     */
+    public void checkAndTriggerSubscriptionManagement(Artefact artefact) {
+        //TODO: fully switch this logic to localdates once artefact model changes
+        if (artefact.getDisplayFrom().toLocalDate().isBefore(LocalDate.now().plusDays(1))
+            && (artefact.getDisplayTo() == null
+            || artefact.getDisplayTo().toLocalDate().isAfter(LocalDate.now().minusDays(1)))) {
+            log.info(sendArtefactForSubscription(artefact));
+        } else {
+            log.error("invalid publication, no trigger sent");
+        }
+    }
+
+    public String sendArtefactForSubscription(Artefact artefact) {
+        //TODO: PUB 1001 check which subscribers are valid here
+        return subscriptionManagementService.sendArtefactForSubscription(artefact);
+    }
```

Comment:
oh wait, youve moved it??? whats the reasoning for hiding this new logic in the middle of the service file?

Reply:
I just put it directly underneath the trigger method, wasn't trying to cause issues.